src/vma/sock/sock-redirect.cpp

```diff
@@ -507,7 +507,12 @@ int vma_get_socket_rings_fds(int fd, int *ring_fds, int ring_fds_sz)
 extern "C"
 int vma_add_conf_rule(char *config_line)
 {
-	do_global_ctors();
+	int res = do_global_ctors();
+
+	if (res) {
+		vlog_printf(VLOG_ERROR, "vma failed to start %d\n", res);
```

Comment:
If res can only be -1 I don't see any point in printing it (same for all other cases in the file).

Just and idea, if you're necessarily handling do_global_ctors() consistently throughout the file, you can consider something like:

#define DO_GLOBAL_CTORS() do { \
        int res = do_global_ctors(); \
        if (res) { \
                vlog_printf(...); \
                return -1; \
        } \
} while (0)